emcc.py

```diff
@@ -2942,18 +2962,28 @@ def un_src(self):
     """Use this if you want to modify the script and need it to be inline."""
     if self.src is None:
       return
+    run_module = ''
+    if shared.Settings.MODULARIZE and not shared.Settings.MODULARIZE_INSTANCE:
+      run_module = 'script.onload = function() { %s(Module); };' % shared.Settings.EXPORT_NAME
```

Comment:
If I build with
```
emcc tests\hello_world.c -s MODULARIZE=1 -o a.html -s EXPORT_NAME=MyModule
```
what should happen? Do we expect to generate a web page that does not run at all? (but user has to open up web console and run `MyModule(Module);` there? That may be a bit odd out of the box behavior. The above change makes it so that a `-o a.html` page compiled with -s MODULARIZE=1 will generate an output that runs automatically.

If developer is targeting a `-o a.js` build, then the above code is avoided, and they can have their "I'll run it manually" feature.

I suppose I am a bit indifferent about this, I can revert this if this may be problematic - just thought I'd make these kind of `-s MODULARIZE=1 -o a.html` builds run out of the box by default.

Reply:
> Do we expect to generate a web page that does not run at all?

Yes, exactly. Only a function `MyModule` should exist in the global scope, and the user is expected to call it in a way that makes sense for their application. Consider that some user may only wany to  call `MyModule` after some special input event, for example. And some users want to call it more than once.

The other behavior, where the code is run, is what `MODULARIZE_INSTANCE` does: it automatically runs it for you (and exactly once). It sounds like you're expecting `MODULARIZE` to work the same, but the use cases are different.

Reply:
MODULARIZE_INSTANCE is different than MODULARIZE in that MODULARIZE_INSTANCE assigns export over to `{{{ EXPORT_NAME }}}` and kills the module function, whereas MODULARIZE returns the exports, and the function can be called many times. I was not expecting MODULARIZE to work like MODULARIZE_INSTANCE, but I thought that `-o a.html` would be good to be able to run out of the box, while retaining the module-like structure.

Reply:
Maybe I still don't get your point here, but "run out of the box" is exactly what modularize is supposed to **not** do? Making it run automatically would be a breaking and surprising change. I don't think that's different for HTML - the user would still be writing code to call the function themselves, so if you call it that would interfere.

Reply:
Gotcha - yeah, then it sounds like the right option is to abandon the autorun when building to html, so this PR no longer has that.